managed/src/main/java/com/yugabyte/yw/models/helpers/CustomerConfigValidator.java

```diff
@@ -229,6 +238,56 @@ protected void doValidate(String value, ObjectNode errorsCollector) {
     }
   }
 
+  public static class ConfigGCSPreflightCheckValidator extends ConfigValidator {
+
+    protected final String fieldName;
+
+    public ConfigGCSPreflightCheckValidator(String type, String name, String fieldName) {
+      super(type, name);
+      this.fieldName = fieldName;
+    }
+
+    @Override
+    public void doValidate(JsonNode data, ObjectNode errorJson) {
+      if (this.name.equals("GCS") && data.get(GCS_CREDENTIALS_JSON_FIELDNAME) != null) {
+        String gsUriPath = data.get(BACKUP_LOCATION_FIELDNAME).asText();
+        String gsUri = gsUriPath;
+        // Assuming bucket name will always start with gs:// otherwise that will be invalid
+        if (gsUriPath.length() < 5 || !gsUriPath.startsWith("gs://")) {
+          errorJson.set(fieldName, Json.newArray().add("Invalid gsUriPath format: " + gsUriPath));
+        } else {
+          gsUriPath = gsUriPath.substring(5);
+          String[] bucketSplit = gsUriPath.split("/", 2);
+          String bucketName = bucketSplit.length > 0 ? bucketSplit[0] : "";
+          String prefix = bucketSplit.length > 1 ? bucketSplit[1] : "";
+          String gcpCredentials = data.get(GCS_CREDENTIALS_JSON_FIELDNAME).asText();
+          try {
+            Credentials credentials =
+                GoogleCredentials.fromStream(
+                    new ByteArrayInputStream(gcpCredentials.getBytes("UTF-8")));
+            Storage storage =
+                StorageOptions.newBuilder().setCredentials(credentials).build().getService();
+            Page<Blob> blobs =
+                storage.list(
+                    bucketName,
+                    Storage.BlobListOption.prefix(prefix),
+                    Storage.BlobListOption.currentDirectory());
+            final List<Blob> blobItems = new ArrayList<>();
```

Comment:
Is it the complete logic and we only need to check that "blobs" has any data? If yes, then we don't need the `blobItems` list and don't need the loop.

Reply:
Yeah but I don't see other way to check whether **blobs** has some **blob** or not.
https://googleapis.dev/java/gax/latest/com/google/api/gax/paging/Page.html
Please suggest! Thanks!

Reply:
Which type has blob.iterateAll()? Does it have `size()`, `hasNext()` or something like this?

Reply:
blob.iterateAll() has java.lang.Iterable<ResourceT> type.


It has getNextPage(), hasNextPage(), getValues(), getNextPageToken() but that don't seem to be working in this case. Already tried them.